Initium-Odp/src/com/universeprojects/miniup/server/domain/NPCDef.java

```diff
@@ -0,0 +1,225 @@
+package com.universeprojects.miniup.server.domain;
+
+import java.util.List;
+
+import com.universeprojects.cacheddatastore.CachedEntity;
+
+//Defines how to generate a new NPC for the character entity type.
+public class NPCDef extends OdpDomain {
+	public static final String KIND = "NPCDef";
+
+	public NPCDef() {
+		super(new CachedEntity(KIND));
+	}
+
+	public NPCDef(CachedEntity cachedEntity) {
```

Comment:
Imo, this would work better as a static method like NPCDef.wrap(cachedEntity). It makes it clear that a new entity is not being created and it's just wrapping.